src/signatures.rs

```diff
@@ -0,0 +1,1206 @@
+    let keypairs = AccountKeys::from(InitialAccountData {
+        keys:      [(KeyIndex(0), keypair)].into_iter().collect(),
+        threshold: SignatureThreshold::ONE,
+    });
+    let signature = sign_as_account(client, signer, keypairs, message, bi).await?;
+    // Accessing the maps at indices 0 is safe because we generated an
+    // AccountSignature with a keypair at index 0 at both maps.
+    Ok(signature.sigs[&0].sigs[&0].clone())
+}
+
+/// Generates account signatures similarly to the [`sign_as_account`] function
+/// but does not perform any checks against the on-chain data such as:
+/// - if the signer account exists on chain.
+/// - if the credential or key indices from the `account_keys` map exist on
+///   chain.
+/// - if the public keys in the `account_keys` map match the public keys on
+///   chain.
+/// - if the public keys in the `account_keys` map match the private keys in the
+///   `account_keys` map.
+pub fn sign_as_account_unchecked(
+    signer: AccountAddress,
+    account_keys: &AccountKeys,
+    message: &Message<'_>,
+) -> AccountSignatures {
+    let message_hash = calculate_message_hash(message, signer);
+
+    // Generate the return type.
+    let mut account_signatures = AccountSignatures {
+        sigs: BTreeMap::new(),
+    };
+
+    for (credential_index, credential) in &account_keys.keys {
+        for (key_index, signing_keys) in &credential.keys {
+            let signature = signing_keys.sign(&message_hash);
+
+            account_signatures
+                .sigs
+                .entry(credential_index.index)
+                .or_insert_with(|| CredentialSignatures {
+                    sigs: BTreeMap::new(),
+                })
+                .sigs
+                .insert(key_index.0, signature.into());
+        }
+    }
+
+    account_signatures
+}
+
+/// Wrapper around the [`sign_as_account_unchecked`] function to simplify the
+/// signature generation for single-signer accounts.
+pub fn sign_as_single_signer_account_unchecked(
+    signer: AccountAddress,
+    signing_key: SigningKey,
+    message: &Message<'_>,
+) -> Result<Signature, SignatureError> {
+    let keypair: KeyPair = KeyPair::from(signing_key);
+    // Generate account keys that have one keypair at index 0 in both maps.
+    let keypairs = AccountKeys::from(InitialAccountData {
+        keys:      [(KeyIndex(0), keypair)].into_iter().collect(),
+        threshold: SignatureThreshold::ONE,
+    });
+    let signature = sign_as_account_unchecked(signer, &keypairs, message);
+    // Accessing the maps at indices 0 is safe because we generated an
+    // AccountSignature with a keypair at index 0 at both maps.
+    Ok(signature.sigs[&0].sigs[&0].clone())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use concordium_base::{base::AccountThreshold, id::types::CredentialData};
+    use std::str::FromStr;
+
+    const NODE_URL: &str = "http://node.testnet.concordium.com:20000";
```

Comment:
So the tests perform networking to this node? I suppose it's fine even if it makes the tests slightly brittle.